rules/implementation.js

```diff
@@ -0,0 +1,66 @@
+const {
+  is
+} = require('bpmnlint-utils');
+
+const implementationAttributes = [
+  'camunda:expression',
+  'camunda:delegateExpression',
+  'camunda:class',
+  'camunda:type'
+];
+
+/**
+ * Rule that reports the usage of collapsed sub-processes.
+ */
+module.exports = function() {
+
+  function check(node, reporter) {
+    if (is(node, 'camunda:ServiceTaskLike')) {
+
+      // quirk: bpmn:MessageEventDefinition is a camunda:ServiceTaskLike
+      // even though it's not a bpmn:Task
+      if (is(node, 'bpmn:MessageEventDefinition')) {
```

Comment:
Why is this one excluded? It does require an implementation to work (if embedded in a throwing message event). Or is it optional from the engine perspective?

Reply:
I've tested this with the engine and it accepts both intermediate send event as well as message end tasks without implementation.

Reply:
Maybe it's a bug on the engine: https://docs.camunda.org/manual/7.15/reference/bpmn20/events/message-events/#camunda-extensions-for-messageeventdefinition

Reply:
Let's be more strict; I'll remove this condition.

Reply:
Could you double check with the Engine folks first? Let's support what is implemented, not what is supposed to be fixed.

If the engine supports "none" message throw events, that is fine for me, too.

Reply:
I'm aware of the fact that this _is_ the way to make these throw events executable. But if no implementation does not result in any error, that should be fine for us, too.

Reply:
Yeah well. My tests revealed that the events were just treated as none events. I'll re-implement the type check and let the engine people know about the _mandatory but not mandatory_ behaviour of the engine.

Reply:
I've started the dicussion [here](https://camunda.slack.com/archives/C6M6X3EKZ/p1618831753091200). I think we can change the rule when/if the engine starts to reject deployments with missing implementation. Until then, we can keep the less strict version.